Initium-Odp/src/com/universeprojects/miniup/server/commands/CommandAutofixDeathModeNotSet.java

```diff
@@ -0,0 +1,87 @@
+package com.universeprojects.miniup.server.commands;
+
+import java.util.Map;
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.datastore.Key;
+import com.universeprojects.cacheddatastore.CachedDatastoreService;
+import com.universeprojects.cacheddatastore.CachedEntity;
+import com.universeprojects.miniup.CommonChecks;
+import com.universeprojects.miniup.server.GameUtils;
+import com.universeprojects.miniup.server.ODPDBAccess;
+import com.universeprojects.miniup.server.WebUtils;
+import com.universeprojects.miniup.server.commands.framework.Command;
+import com.universeprojects.miniup.server.commands.framework.UserErrorMessage;
+import com.universeprojects.miniup.server.commands.framework.Command.JavascriptResponse;
+
+/**
+ * Checks to see if a character has non-positive HP but did not correctly have its mode changed to
+ * "Dead" or "Unconscious". If so, set mode to unconscious and clear equipment & combatant. If character
+ * is in a RestSite location, then trigger their revival. 
+ * 
+ * @author papamarsh
+ * 
+ */
+public class CommandAutofixDeathModeNotSet extends Command {
+
+	public CommandAutofixDeathModeNotSet(ODPDBAccess db, HttpServletRequest request,
+			HttpServletResponse response) {
+		super(db, request, response);
+	}
+
+	@Override
+	public void run(Map<String, String> parameters) throws UserErrorMessage {
+		ODPDBAccess db = ODPDBAccess.getInstance(request);
+		CachedDatastoreService ds = getDS();
+		
+		//grab the current character and its location
+		CachedEntity character = db.getCurrentCharacter();
+		CachedEntity currentLocation = db.getEntity((Key) character.getProperty("locationKey"));
+		
+		//grab the character's current hitpoints and mode
+		Double hitpoints = (Double) character.getProperty("hitpoints");
+		String mode = (String) character.getProperty("mode");
+		
+		//check for non-positive HP while mode is not either unconscious or dead
+		if(hitpoints<=0 && mode!="UNCONSCIOUS" && mode!="DEAD")

```

Comment:
Strings cannot be compared this way in java. use .equals(). mode can also be null, so use "UNCONSCIOUS".equals(mode) to avoid additional null checks.